src/python/pants/invalidation/cache_manager.py

```diff
@@ -172,20 +172,34 @@ def __init__(self, cache_manager, target, cache_key):
     super(VersionedTarget, self).__init__(cache_manager, [self])
     self.id = target.id
 
+  def _results_dir_prefix(self, root_dir):
```

Comment:
Hi - sorry for the late reply - I got sick as a dog this week.

I did not give you a great link w/r/t to refactor itself. The changes and reasoning behind it should be mostly complete in this commit: https://github.com/mateor/pants/commit/3c1e2390cfee0a966c353a8b02d619f65125506b.

The refactor was broken out of the work I did [wiping invalid results_dirs.](https://github.com/pantsbuild/pants/commit/6a43414801b44ae953a2a45af12c084660e2f7eb.) I ended up reducing the scope because the existing test coverage was a bit light and if I added any bad behavior it could have been very challenging to track.

But the refactor was done because I thought the existing code was pretty hard to reason about. 
* Variable names are misleading
* Relies heavily on mutating properties
* Some methods have *very* surprising side effects ([deleting this line](https://github.com/pantsbuild/pants/commit/6a43414801b44ae953a2a45af12c084660e2f7eb#diff-bb38e789cc53840e6274c13751f6c034L486) removed ~400 file copies *per cache hit* in our repo) .

There was just a lot of older code there. But as evidenced by your question - without a change to hang it on, the refactor somewhat lost its purpose. If you do [take a look](https://github.com/mateor/pants/commit/3c1e2390cfee0a966c353a8b02d619f65125506b) and think I should pick it back up, let me know.